packages/devtools_app/lib/src/screens/logging/logging_controller.dart

```diff
@@ -548,40 +578,49 @@ class LoggingController extends DisposableController
   @override
   void filterData(Filter<LogData> filter) {
     super.filterData(filter);
-    final queryFilter = filter.queryFilter;
-    if (queryFilter.isEmpty) {
-      filteredData
-        ..clear()
-        ..addAll(data);
-      return;
-    }
-    filteredData
-      ..clear()
-      ..addAll(
-        data.where((log) {
-          final kindArg = filter.queryFilter.filterArguments[kindFilterId];
-          if (kindArg != null &&
-              !kindArg.matchesValue(log.kind.toLowerCase())) {
-            return false;
-          }
 
-          if (filter.queryFilter.substrings.isNotEmpty) {
-            for (final substring in filter.queryFilter.substrings) {
-              final matchesKind = log.kind.caseInsensitiveContains(substring);
-              if (matchesKind) return true;
+    bool filterCallback(LogData log) {
+      for (final toggleFilter in filter.toggleFilters) {
+        if (toggleFilter.enabled.value && !toggleFilter.includeCallback(log)) {
+          return false;
+        }
+      }
+
+      final queryFilter = filter.queryFilter;
+      if (!queryFilter.isEmpty) {
+        final kindArg = queryFilter.filterArguments[kindFilterId];
+        if (kindArg != null &&
```

Comment:
Ditto for making this a variable

Reply:
Added a TODO. I will address in a follow up PR to clean up the query filter arguments in general.